File: HelpSoftware/Help/Help.EF/Entitys/SearchHistory.cs
﻿using System;
using System.ComponentModel.DataAnnotations;

namespace Help.EF
{
  public class SearchHistory : IHelpEntity
  {
    [Key]
    public virtual int ID_SearchHistory { get; set; }

    public virtual string SearchText { get; set; }

    public virtual DateTime SearchTime { get; set; }

    public virtual User SearchedUser { get; set; }

    [Timestamp]
    public virtual byte[ ] TimeStamp { get; set; }

    public string GetEntityName()
    {
      return "SearchHistory";
    }

    public int GetID()
    {
      return ID_SearchHistory;
    }

    public bool GetWriteHistoryEntry()
    {
      return false;
    }
  }
}

File: HelpSoftware/Help/Help.Web/Web/src/app/EF/SearchHistory.ts
import { User } from "./EF";

export class SearchHistory {
  ID_SearchHistory: number;
  SearchText: string;
  SearchTime: Date;
  SearchedUser: User;

  deserialize(input): SearchHistory {
    this.ID_SearchHistory = input.ID_SearchHistory;
    this.SearchText = input.SearchText;
    this.SearchTime = input.SearchTime;
    this.SearchedUser = new User().deserialize(input.SearchedUser);
    return this;
  }
}
